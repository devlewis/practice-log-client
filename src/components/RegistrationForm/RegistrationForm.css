.red {
  color: red;
}

.RegistrationForm {
  background-color: aliceblue;
  padding: 10px 10px 10px 10px;
  width: 300px;
  margin: auto;
  display: flex;
  flex-direction: column;
}

.RegistrationForm h2 {
  margin: 20px 20px;
}

.user_name input,
.password input {
  margin-top: 5px;
  border: 1px solid gray;
}

.user_name,
.password {
  margin: 10px 0;
}
.restrbuttonrow {
  margin: 20px 0 10px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}
