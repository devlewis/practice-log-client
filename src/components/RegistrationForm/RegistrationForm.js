import React, { Component } from "react";
import AuthApiService from "../../services/auth-api-service";

export default class RegistrationForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  };

  state = { error: null, user_name: "", password: "" };

  handleRegistrationSuccess = user => {
    const { history } = this.props;
    history.push("/");
  };

  handleSubmit = ev => {
    ev.preventDefault();
    const { user_name, password } = this.state;

    this.setState({ error: null });
    AuthApiService.postUser({
      user_name: user_name,
      password: password
    })
      .then(user => {
        this.setState({ user_name: "", password: "" });
        this.handleRegistrationSuccess();
      })
      .catch(res => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error } = this.state;
    return (
      <form className="RegistrationForm" onSubmit={this.handleSubmit}>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <div className="user_name">
          <label htmlFor="RegistrationForm__user_name">User name</label>
          <input
            name="user_name"
            type="text"
            required
            id="RegistrationForm__user_name"
            onChange={e => this.setState({ user_name: e.target.value })}
          ></input>
        </div>
        <div className="password">
          <label htmlFor="RegistrationForm__password">Password</label>
          <input
            name="password"
            type="password"
            required
            id="RegistrationForm__password"
            onChange={e => this.setState({ password: e.target.value })}
          ></input>
        </div>
        <button type="submit">Register</button>
      </form>
    );
  }
}
