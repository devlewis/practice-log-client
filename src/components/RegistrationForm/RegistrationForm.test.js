import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RegistrationForm from "./RegistrationForm";

describe(`RegistrationForm component`, () => {
  const props = {
    className: "test-class-name",
    children: <p>test children</p>,
    "data-other": "test-other-prop",
  };

  it("renders form.RegistrationForm by default", () => {
    const wrapper = shallow(<RegistrationForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the RegistrationForm given props", () => {
    const wrapper = shallow(<RegistrationForm {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
