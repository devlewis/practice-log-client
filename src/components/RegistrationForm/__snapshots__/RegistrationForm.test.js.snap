// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm component renders form.RegistrationForm by default 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm__user_name"
    >
      User name
    </label>
    <input
      id="RegistrationForm__user_name"
      name="user_name"
      onChange={[Function]}
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Password
    </label>
    <input
      id="RegistrationForm__password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
    />
  </div>
  <button
    type="submit"
  >
    Register
  </button>
  <button
    onClick={[Function]}
  >
    Cancel
  </button>
</form>
`;

exports[`RegistrationForm component renders the RegistrationForm given props 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm__user_name"
    >
      User name
    </label>
    <input
      id="RegistrationForm__user_name"
      name="user_name"
      onChange={[Function]}
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Password
    </label>
    <input
      id="RegistrationForm__password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
    />
  </div>
  <button
    type="submit"
  >
    Register
  </button>
  <button
    onClick={[Function]}
  >
    Cancel
  </button>
</form>
`;
